src/NuGet.Core/NuGet.ProjectModel/ProjectLockFile/PackageInstallationType.cs

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace NuGet.ProjectModel
+{
+    public enum PackageInstallationType
```

Comment:
I don't think this is a good name. 
Installation type transitive sounds weird to me. 

How about PackageDependencyType?

Reply:
I'm ok with `PackageDependencyType` if that's not already taken